freedom/neural_nets/transformations.py

```diff
@@ -41,11 +41,63 @@ def __init__(self, labels, min_energy=0.1, max_energy=1e4, use_pmt_dir=False):
         #self.e_cscd_scale = self.add_weight(shape=(1,), initializer="zeros", trainable=True)
         #loc_x_bias
         
-        
+
     def get_config(self):
         return {'labels': self.labels, 'max_energy': self.min_energy, 'max_energy': self.max_energy, 
                 'use_pmt_dir': self.use_pmt_dir}
 
+
+    def TimeResidual(self, hits, params, Index=1.33):
```

Comment:
these are now defined twice, once here and once in utils?

Reply:
So far yes. Ideally they would just be defined here (where we have to calculate them anyway) and then passed to the reco. But this is not so straight forward to do and so the (numpy) definition in utils is a workaround for the moment.